src/decisionengine_modules/NERSC/sources/NerscSFApi.py

```diff
@@ -62,43 +61,51 @@ def check_accesstoken(self, nersc_user):
         except KeyError:
             self.logger.error(f"Unknown user '{nersc_user}', exiting")
             return None
-
+        print(nersc_user)
         rawfile = params_dict["rawfile"]
         pemfile = params_dict["private_key"]
         clientidfile = params_dict["client_id_file"]
         with open(clientidfile) as cifile:
             client_id = cifile.read()
             client_id = client_id.rstrip()
-
+        atoken = None
         if not os.path.exists(rawfile):
             self.logger.debug(f"{rawfile} does not exist. Need to generate")
         else:
             atoken = None
             with open(rawfile) as afile:
                 atoken = afile.read()
                 atoken = atoken.rstrip()
-            # HK> If the access token is expired, the flow goes directly to except jwt.ExpiredSignatureError
-            try:
-                jwt.decode(atoken, options={"verify_signature": False})
-                self.logger.debug("Access Token not expired. Returning without generating a new access token")
-                return atoken  # This means the existing access token is not expired.
-
-            except jwt.ExpiredSignatureError:
-                self.logger.debug("Access Token expired")
-
-        certs = pem.parse_file(pemfile)
-        private_key = str(certs[0])
-        client = OAuth2Session(
-            client_id=client_id, client_secret=private_key, token_endpoint_auth_method="private_key_jwt"
-        )
-        client.register_client_auth_method(PrivateKeyJWT(token_url))
-        resp = client.fetch_token(token_url, grant_type="client_credentials")
+            # HK> If the access token is expired, the flow goes directly to except jwt.ExpiredSign
 
-        newtoken = resp["access_token"]
+        if atoken is not None:
+            rvalue = jwt.decode(atoken, options={"verify_signature": False})
+            ctime = int(time.time())
+            diff = ctime - rvalue["exp"]
+            print(diff)
+        else:
```

Comment:
There was a reason we had to do it this way.  There are certain cases the diff ends up undefined.  will look at the code and get back to you on this.

Reply:
See the above comment.  If adding that extra option to `jwt.decode` works, and raises the right error, then you won't need to calculate the diff.  Otherwise, OK.

Reply:
So the problem is the following--if the token file doesn't exist at all (which at startup it never does)
then you have no token and you can't calculate a diff, so you have to assign a diff somehow.
That's why I had to move the if statement out of the if-else clause above,  because there are a number
of cases where the token doesn't exist at all and diff would be undefined unless you set it to somel
I agree the above code is awkward but don't see an easy way around it.  Even the verify_exp: True
thing you suggest above doesn't help that.  Given those constraints--if you have any ideas on how to streamline the code I am all ears.. I realize this is not a very smooth logical code flow but don't know how to get around it easily.

Reply:
Were you able to review these options?  I'm fine with keeping things as you have them in the PR, but it seems to me like it could be clearer if we use the new API.  I won't push on this point, as functionally, your code looks good to me.

Reply:
So I think the only thing to focus on for this PR is:

1. Do we want to keep it as you have it in the PR now, which is a bit more direct, doesn't use the new API, but probably will not break if PyJWT decides to change the API without declaring a major change; or
2. Do we try to use the `verify_exp` argument/API, and most of the new code you have here wouldn't be necessary (see the "Not using diff" section in my discussion comment above).